linera-core/src/chain_worker/actor.rs

```diff
@@ -147,220 +146,184 @@ where
     },
 }
 
-/// The actor worker type.
-pub struct ChainWorkerActor<StorageClient>
-where
-    StorageClient: Storage + Clone + Send + Sync + 'static,
-    ViewError: From<StorageClient::StoreError>,
-{
-    worker: ChainWorkerState<StorageClient>,
-    incoming_requests: mpsc::UnboundedReceiver<ChainWorkerRequest<StorageClient::Context>>,
-    service_runtime_thread: JoinHandle<()>,
+/// Spawns a blocking task to execute the service runtime actor.
```

Comment:
I don't think there's a need to refactor everything here :/

Reply:
I moved the logic of from `spawn` into `run` in order to handle the setup in that task.  But then Clippy complains `ChainWorkerActor` is never constructed.

If we're going to move the state `load` out into `WorkerState`, this refactor doesn't technically need to happen any more.  But Clippy is right: since the only thing you can do with a constructed `ChainWorkerActor` is immediately call `run` on it, there's not much point in keeping the functions separate, and letting the compiler auto-generate the state layout makes modifications easier in the future.  Particularly if we're going to rename `ChainWorkerState` to `ChainWorker`, I think we should keep this change, as otherwise the names `ChainWorker` and `ChainWorkerActor` (semantically) conflict.

Reply:
I've removed this commit for now.